scipy/special/ellint_carlson_cpp_lite/ellint_arith_aux.hh

```diff
@@ -0,0 +1,195 @@
+#ifndef ELLINT_ARITH_AUX_HH_INCLUDED
+#define ELLINT_ARITH_AUX_HH_INCLUDED
+
+
+#include <cstddef>
+#include "ellint_typing.hh"
+
+
+/* Auxiliary floating-point manipulation utilities.
+ * Ref:
+ *
+ * S. M. Rump, T. Ogita, S. Oishi: Accurate Floating-Point Summation.
+ *      2005-11-13, Tech. Rep. 05.1, Fac. Inf. Commun. Sci, Hambg. Univ.
+ *      Technol., also known as SIAM J. Sci. Comput. Volume 31, Issue 1, pp.
+ *      189-224 (2008),
+ *      http://www.ti3.tuhh.de/paper/rump/RuOgOi07I.pdf
+ *
+ * The main purpose of these templates is to implement Algorithm 6.1 (here as
+ * acc_sum) which is required by compensated complex floating-point polynomial
+ * evaluation.
+ */
+namespace ellint_carlson { namespace arithmetic { namespace aux
+{
+    /* Algorithm 3.6 */
+    template<typename RT>
+    inline typing::real_only<RT, RT>
+    next_power_two(const RT& p)
+    {
+	RT q = p / typing::half_epsilon<RT>();
+	RT L = std::abs((q + p) - q);
+	if ( L == 0.0 )
```

Comment:
Some places in this code follow the convention `if (literal [op] variable)` is followed, but here is an example of where it is not.  While it's not in PEP7 (I thought it was, but didn't see it when I checked), choosing the literal first will allow the compiler to catch mistakes of the form `if (variable = literal)`.  If you're going back anyways, flipping the order is an easy way to more safety for (almost) free.  Again, not something I'll hold the PR up for